### PR TITLE
feat: add pod disruption budget and update app version

### LINKED ISSUE #80: Chart should have a PDB for production deployments

### PR DESCRIPTION
**Pull request by bitfactory-sem-denbroeder**
closes #80

### CODE REVIEW

**Comment by hifabienne** [comment]
@bitfactory-sem-denbroeder Thanks for your contribution.
@eliobischof can you have a look at it?

File: charts/zitadel/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: zitadel
 description: A Helm chart for ZITADEL v2
 type: application
-appVersion: "v2.21.1"
+appVersion: "v2.23.1"
```
Comment (eliobischof) [inline charts/zitadel/Chart.yaml:5]:
```suggestion
appVersion: "v2.22.0"
```

File: charts/zitadel/values.yaml [hunk 2]
```diff
@@ -175,3 +175,7 @@ metrics:
     enabled: false
     honorLabels: false
     honorTimestamps: true
+
+pdb:
+  enabled: false
+  minAvailable: 1
```
Comment (eliobischof) [inline charts/zitadel/values.yaml:181]:
```suggestion
``````suggestion
```

File: charts/zitadel/templates/pdb.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{{- if .Values.pdb.enabled }}
+{{- if .Values.pdb.apiVersion }}
+apiVersion: {{ .Values.pdb.apiVersion }}
+{{- else }}
+apiVersion: policy/v1beta1
+{{- end }}
```
Comment (eliobischof) [inline charts/zitadel/templates/pdb.yaml:6]: Can we handle this like we do in the ingress template?

File: charts/zitadel/templates/pdb.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+kind: PodDisruptionBudget
+metadata:
+  name: {{ include "zitadel.fullname" . }}
+  labels:
+    {{- include "zitadel.labels" . | nindent 4 }}
```
Comment (eliobischof) [inline charts/zitadel/templates/pdb.yaml:11]: labels annotations as well like in the ingress template

File: charts/zitadel/templates/pdb.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+spec:
+  {{- if .Values.pdb.minAvailable }}
+  minAvailable: {{ .Values.pdb.minAvailable }}
+  {{- end }}
+  {{- if .Values.pdb.maxUnavailable }}
+  maxUnavailable: {{ .Values.pdb.maxUnavailable }}
+  {{- end }}
+  selector:
```
Comment (eliobischof) [inline charts/zitadel/templates/pdb.yaml:19]: Is it possible to configure 0 for minAvailable? Because 0 is the go zero value, right?

**Review by eliobischof — CHANGES_REQUESTED** [review]
Thanks a lot for the PR :tada:

File: charts/zitadel/templates/pdb.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  name: {{ include "zitadel.fullname" . }}
+  labels:
+    {{- include "zitadel.labels" . | nindent 4 }}
+spec:
+  {{- if .Values.pdb.minAvailable }}
+  minAvailable: {{ .Values.pdb.minAvailable }}
+  {{- end }}
+  {{- if .Values.pdb.maxUnavailable }}
+  maxUnavailable: {{ .Values.pdb.maxUnavailable }}
+  {{- end }}
```
Comment (eliobischof) [inline charts/zitadel/templates/pdb.yaml:18]:
```suggestion
  minAvailable: {{ .Values.pdb.minAvailable }}
  maxUnavailable: {{ .Values.pdb.maxUnavailable }}
```

**Review by eliobischof — CHANGES_REQUESTED** [review]

File: charts/zitadel/templates/pdb.yaml (hunk 6 above)
Comment (bitfactory-sem-denbroeder) [reply charts/zitadel/templates/pdb.yaml]: Yep, looks better

File: charts/zitadel/templates/pdb.yaml (hunk 4 above)
Comment (bitfactory-sem-denbroeder) [reply charts/zitadel/templates/pdb.yaml:11]: Nice one! missed that

File: charts/zitadel/templates/pdb.yaml (hunk 3 above)
Comment (bitfactory-sem-denbroeder) [reply charts/zitadel/templates/pdb.yaml]: That's fancy! Unsure if this is the correct way to implement, but I tried nonetheless

**Comment by bitfactory-sem-denbroeder** [comment]
@eliobischof Updated your changes! Hope it's good now.

File: charts/zitadel/Chart.yaml [hunk 7]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: zitadel
 description: A Helm chart for ZITADEL v2
 type: application
-appVersion: "v2.21.1"
+appVersion: "v2.22.0"
 version: 4.2.0
```
Comment (eliobischof) [inline charts/zitadel/Chart.yaml:6]: Sorry, suggested it wrong before :slightly_smiling_face: 

```suggestion
appVersion: "v2.23.1"
version: 4.3.0
```

**Review by eliobischof — CHANGES_REQUESTED** [review]

**Comment by eliobischof** [comment]
Looks almost good to me, thank you :partying_face:
We'd love to send you some swag. If you like that too, please send us a mail to hi@zitadel.com.

**Review by eliobischof — APPROVED** [review]

**Comment by bitfactory-sem-denbroeder** [comment]
Updated the PDB as the tests were failing, hopefully it will all be alright now. Added back the previous check for pdb max and min available and added a helm failure state. If this is not desired, I can change it back. 

(pushed the changes from my personal PC, that's why the author is different. both accounts are me 😄 )

**Comment by eliobischof** [comment]
> Updated the PDB as the tests were failing, hopefully it will all be alright now. Added back the previous check for pdb max and min available and added a helm failure state. If this is not desired, I can change it back.
> 
> (pushed the changes from my personal PC, that's why the author is different. both accounts are me smile )

Thx, LGTM

**Review by eliobischof — APPROVED** [review]

**Pull request merged**